projects/pages/xhr/Login-xhr.jsx

```diff
@@ -0,0 +1,94 @@
+import { useRouter } from 'next/router'
+import { useState } from 'react';
+
+export default function LoginXHR() {
+    const router = useRouter();
+    const [ data, setData ]= useState({});
+    
+    const handleSubmit = async (e) => {
+      e.preventDefault();
+      const http = new XMLHttpRequest();
+      const url = "https://staging.komunitasmea.com/api/login";
+      let formBody = [];
+      for (let property in data) {
+          let encodedKey = encodeURIComponent(property);
+          let encodedValue = encodeURIComponent(data[property]);
+          formBody.push(encodedKey + "=" + encodedValue);
+      }
+      formBody = formBody.join("&");
+      http.open('POST', url, true);
+      http.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+      http.responseType= 'json';
+      http.withCredentials = true;
+      http.onload = function() {
+        if(http.readyState == 4 && http.status == 200) {
```

Comment:
kenapa ini harus ready state 4 ?

Reply:
UNSENT = 0; // initial state
OPENED = 1; // open called
HEADERS_RECEIVED = 2; // response headers received
LOADING = 3; // response is loading (a data packet is received)
DONE = 4; // request complete

mengacu dokumen, readystate ===4 karena request done atau telah selesai